projects/OG-Financial/src/main/java/com/opengamma/financial/analytics/model/fixedincome/CashFlowDetailsProvider.java

```diff
@@ -69,28 +75,47 @@
    * @param definition the SwapDefinition containing the payment definitions
    * @param security the InterestRateSwapSecurity
    * @param type the PayReceiveType, either pay or receive
+   *
+   * @deprecated use constructor that takes 'full' flag to specify whether to include past cash flows
    */
+  @Deprecated
   public CashFlowDetailsProvider(MulticurveProviderInterface multicurveProviderInterface,
                                  ZonedDateTime zonedDateTime,
                                  SwapDefinition definition,
                                  InterestRateSwapSecurity security,
                                  PayReceiveType type) {
-    ArgumentChecker.notNull(multicurveProviderInterface, "multicurveProviderInterface");
-    ArgumentChecker.notNull(zonedDateTime, "zonedDateTime");
-    ArgumentChecker.notNull(definition, "definition");
-    ArgumentChecker.notNull(type, "type");
```

Comment:
Are these ArgumentChecker valudations no longer useful? In the overloaded constructor we could still check for not null.